### PR TITLE
ISSUE-913 Implement postgres jmap settings repository

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

**Comment by quantranhong1999** [comment]
[ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.6_0.9.34:scalafix (scala-check-style) on project jmap-extensions: Scalafix invoked with errors. Check logs for details. -> [Help 1]

File: tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/settings/PostgresJmapSettingsDAO.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    public Mono<Void> updateSettings(Username username,
+                                     UuidState newState,
+                                     Map<JmapSettingsKey, JmapSettingsValue> addSettings,
+                                     List<JmapSettingsKey> removeSettings) {
+        return postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.update(TABLE_NAME)
+            .set(STATE, newState.value())
+            .set(DSL.field(SETTINGS.getName()),
+                (Object) DSL.field("delete(" + SETTINGS.getName() + " || ?, ?)",
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/settings/PostgresJmapSettingsDAO.java:103]: This string concatenation in a sql query is more than likely to be prone to a SQL injection.

**Comment by Arsnael** [comment]
```
[ERROR] Failed to execute goal io.github.evis:scalafix-maven-plugin_2.13:0.1.6_0.9.34:scalafix (scala-check-style) on project jmap-extensions: Scalafix invoked with errors. Check logs for details. -> [Help 1]
```

File: tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/settings/PostgresJmapSettingsDAO.java (hunk 1 above)
Comment (hungphan227) [reply tmail-backend/jmap/extensions-postgres/src/main/java/com/linagora/tmail/james/jmap/settings/PostgresJmapSettingsDAO.java:103]: Don't worry. SETTINGS.getName() is constant

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**